src/Windowing/OpenToolkit.Windowing.Desktop/DpiCalculator.cs

```diff
@@ -0,0 +1,206 @@
+            return success;
+        }
+
+        /// <summary>
+        /// Gets the raw dpi of the monitor pointed to.
+        /// </summary>
+        /// <param name="monitor">The monitor in question.</param>
+        /// <param name="horizontalDpi">Horizontal dpi.</param>
+        /// <param name="verticalDpi">Vertical dpi.</param>
+        /// <returns><c>true</c>, if monitor's raw dpi was gotten correctly, <c>false</c> otherwise.</returns>
+        /// <remarks>
+        /// This method calculates dpi by retrieving monitor dimensions and resolution.
+        /// However on certain platforms (such as Windows) these values may not
+        /// be scaled correctly.
+        /// </remarks>
+        public static unsafe bool TryGetMonitorDpiRaw(Monitor* monitor, out float horizontalDpi, out float verticalDpi)
+        {
+            horizontalDpi = verticalDpi = GetPlatformDefaultDpi();
+
+            if (monitor == null)
+            {
+                return false;
+            }
+
+            int widthInMm, heightInMm;
+            int widthInPixels, heightInPixels;
+
+            // According to GLFW documentation, physical dimensions are reported in millimeters.
+            GLFWProvider.GLFW.Value.GetMonitorPhysicalSize(monitor, out widthInMm, out heightInMm);
+
+            var videoMode = GLFWProvider.GLFW.Value.GetVideoMode(monitor);
+
+            if (videoMode == null)
+            {
+                return false;
+            }
+
+            widthInPixels = videoMode->Width;
+            heightInPixels = videoMode->Height;
+
+            if (widthInMm <= 0 || heightInMm <= 0 || widthInPixels <= 0 || heightInPixels <= 0)
+            {
+                return false;
+            }
+
+            /*
+             * How does this formula calculate dpi?
+             *
+             * 1. 1" = 25.4mm thus dInMm / 25.4 = dInInches
+             * 2. dpi = pixelCount/dInInches
+             *
+             * Ergo:
+             *     pixelCount / (dInMm / 25.4)
+             *   = (pixelCount / dInMm) * 25.4
+             */
+
+            horizontalDpi = ((float)widthInPixels / (float)widthInMm) * 25.4f;
+            verticalDpi = ((float)heightInPixels / (float)heightInMm) * 25.4f;
+            return true;
+        }
+
+        /// <summary>
+        /// Gets the default dpi for platforms.
+        /// </summary>
+        /// <returns>The platform default dpi.</returns>
+        /// <remarks>
+        /// For historical reasons macOS has a default dpi of 72, and other
+        /// platforms have a default dpi of 96.
+        /// </remarks>
+        public static float GetPlatformDefaultDpi()
+        {
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.OSX))
+            {
+                return 72f;
+            }
+            else
+            {
+                return 96f;
+            }
+        }
+
+        private static unsafe Rectangle[] GetMonitorClientAreas(Monitor** monitors, int monitorCount)
```

Comment:
Would really like to have that with less allocations. As functions like glfwGetWindowPos can only be called from the main thread anyway we can cache that array in a local static variable without problems. So as long as monitorCount doesn't change we don't need to resize the array....

Reply:
Wouldn't it be better to provide a version that takes in a `Span<Rectangle>` then?

No allocations and also no akward "you do not own this array" stuff.

Reply:
well was thinking about that as, but I'm not sure because rectangle is 16bytes and an array of that on the stack. Best solution would probably be a method which takes in previous monitor rectangle and spits out a single rectangle. Then we would even have less branching :D